### PR TITLE
[Enhancement] Control parquet file size when general compression is applied

### PR DESCRIPTION
**Pull request by letian-jiang**
## What I'm doing:

Control unloaded parquet file size when general compression (e.g. gzip) is applied. 

This PR also fixes some compile warning in main branch.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/parquet_writer.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ Status RollingAsyncParquetWriter::init() {
     ASSIGN_OR_RETURN(auto compression_codec,
                      parquet::ParquetBuildHelper::convert_compression_type(_table_info.compress_type));
     builder.compression(compression_codec);
-    builder.version(::parquet::ParquetVersion::PARQUET_2_0);
```
Comment (letian-jiang) [inline be/src/exec/parquet_writer.cpp:51]: fix compile warning

**Review by dirtysalt — DISMISSED** [review]

**Review by satanson — DISMISSED** [review]

File: be/src/formats/parquet/chunk_writer.cpp [hunk 2]
```diff
@@ -84,7 +84,7 @@ int64_t ChunkWriter::estimated_buffered_bytes() const {
     }
 
     auto buffered_bytes = std::accumulate(_estimated_buffered_bytes.begin(), _estimated_buffered_bytes.end(), 0);
-    return _rg_writer->total_bytes_written() + _rg_writer->total_compressed_bytes() + buffered_bytes;
+    return _rg_writer->total_compressed_bytes_written() + _rg_writer->total_compressed_bytes() + buffered_bytes;
```
Comment (zombee0) [inline be/src/formats/parquet/chunk_writer.cpp:87]: as we use buffered rowgroup writer, is there any difference?

**Review by zombee0 — APPROVED** [review]

File: be/src/formats/parquet/chunk_writer.cpp (hunk 2 above)
Comment (letian-jiang) [reply be/src/formats/parquet/chunk_writer.cpp:87]: yes. the new method measures bytes after compression.

**Review by dirtysalt — APPROVED** [review]

**Comment by letian-jiang** [comment]
@Mergify backport branch-3.3

**Pull request merged**